src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -527,6 +528,9 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                 if (analyzerCts != null)
                 {
                     analyzerCts.Cancel();
+
+                    // Clear cached analyzer descriptors and unregister exception handlers hooked up to the LocalizableString fields of the associated descriptors.
+                    analyzerManager.ClearCachedAnalyzerDescriptors(analyzers);
```

Comment:
For descriptor it does, but LocalizableString has an OnException event which might hold onto things for ever